### PR TITLE
[prometheus-pushgateway] Make path configurable in service monitor

### LINKED ISSUE #3365: [prometheus-pushgateway] serviceMonitor doesn't work if `--web.route-prefix` is set
### Describe the bug a clear and concise description of what the bug is.

Hello prometheus guys 👋🏼,
When you deploy `prometheus-pushgateway` with `--web.route-prefix` parameter, for example `--web.route-prefix=http://localhost/pushgateway`, then deployed serviceMonitor doesn't work properly and gets 404 error.  
This means no related metrics are collected via that serviceMonitor.

### What's your helm version?

v3.10.2

### What's your kubectl version?

v1.25.7

### Which chart?

prometheus-pushgateway

### What's the chart version?

2.1.6

### What happened?

Prometheus web UI display "server returned HTTP status 404 Not Found" error for this specific serviceMonitor.
This means no metrics are collected from _pushgateway_.

### What you expected to happen?

Metrics are correctly scraped via Prometheus.

### How to reproduce it?

Simple reproducer is to run `$ ./pushgateway --web.route-prefix=/pushgateway` and then try `curl localhost:9091/metrics` which will leads to 404 error = no metrics.

### Enter the changed values of values.yaml?

```
spec:
  values:
    extraArgs:
      - --web.route-prefix=/pushgateway
```

### Enter the command that you execute and failing/misfunctioning.

Change above + commands bellow:
```
kubectl port-forward -n monitoring svc/prometheus-pushgateway 7777:http
curl localhost:7777/metrics
```

### Anything else we need to know?

The issue is probably related to this part https://github.com/prometheus-community/helm-charts/blob/88ecb2468b5d1a42d7228e797d137ca74b4dfa6e/charts/prometheus-pushgateway/templates/servicemonitor.yaml#L35 which is not customizable and doesn't allow user to reflect their `--web.route-prefix` setting.

### PR DESCRIPTION
**Pull request by zeritti**
#### What this PR does / why we need it

This PR makes the telemetry path in service monitor configurable through `serviceMonitor.telemetryPath` to support changes via start-up flags like `--web.route-prefix` and `--web.telemetry-path`. The default value remains `/metrics`.

#### Which issue this PR fixes

- fixes #3365

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zanhsieh — APPROVED** [review]

**Comment by QuentinBisson** [comment]
@monotek or @gianrubio can you take a look so this can get merged?

File: charts/prometheus-pushgateway/README.md [hunk 1]
```diff
@@ -41,6 +41,10 @@ helm upgrade [RELEASE_NAME] [CHART] --install
 
 _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
 
+### To 2.2
+
+Field `serviceMonitor.telemetryPath` (default `/metrics`) can now be configured to support changes to the telemetry path via start-up flags.
```
Comment (monotek) [inline charts/prometheus-pushgateway/README.md:46]: imho this is not needed as this section is for major version changes

File: charts/prometheus-pushgateway/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -32,7 +32,7 @@ spec:
     {{- with .Values.serviceMonitor.scrapeTimeout }}
     scrapeTimeout: {{ . }}
     {{- end }}
-    path: /metrics
+    path: {{ default "/metrics" .Values.serviceMonitor.telemetryPath }}
```
Comment (monotek) [inline charts/prometheus-pushgateway/templates/servicemonitor.yaml:35]:
```suggestion
    path: {{ .Values.serviceMonitor.telemetryPath }}
```

File: charts/prometheus-pushgateway/values.yaml [hunk 3]
```diff
@@ -209,6 +209,11 @@ serviceMonitor:
   enabled: false
   namespace: monitoring
 
+  # telemetryPath: HTTP resource path from which to fetch metrics.
+  # Telemetry path, default /metrics, has to be prefixed accordingly if pushgateway sets a route prefix at start-up.
+  #
+  telemetryPath: ""
```
Comment (monotek) [inline charts/prometheus-pushgateway/values.yaml:215]:
```suggestion
  telemetryPath: "/metrics"
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-pushgateway/README.md (hunk 1 above)
Comment (zeritti) [reply charts/prometheus-pushgateway/README.md:46]: Thank you, I will change that.

**Review by monotek — APPROVED** [review]

**Pull request merged**